src/Opdex.Platform.Infrastructure/PlatformInfrastructureServiceCollectionExtensions.cs

```diff
@@ -384,6 +388,15 @@ private static void AddCirrusServices(IServiceCollection services, CirrusConfigu
         services.AddTransient<IRequestHandler<CallCirrusCreateSmartContractCommand, Sha256>, CallCirrusCreateSmartContractCommandHandler>();
     }
 
+    private static void AddCoinGeckoServices(IServiceCollection services)
+    {
+        services.AddTransient<ICoinGeckoClient, CoinGeckoClient>(_ => CoinGeckoClient.Instance);
```

Comment:
I'd have to check but maybe you know, does this have a retry policy? 

We'd also probably want to include a circuit breaker policy too. Circuit breakers after a # of 500 errors will throw exceptions we can alert on, back off for # seconds, then open back up again.

Reply:
Doesn't look to be possible as is, I've forked and amended it so it's possible to configure policies. Opened a PR  https://github.com/tosunthex/CoinGecko/pull/37#issue-1100299073

Reply:
ok this is fine for now then. When we come back to this for price implementation, we may use our own client vs their package then until they support it. Retry and Circuit breakers are pretty important for resiliency